### DISCUSSION TITLE
Settings of motive when using optitrack

### DISCUSSION THREAD

**Discussion by Sherry-97** [USC-ACTLab/crazyswarm#431] [Q&A]
When I use optitrack as a motion capture tool, I test it using [https://github.com/whoenig/NatNetSDKCrossplatform](url) and set it as follows.
![Selection_008](https://user-images.githubusercontent.com/70084772/126453384-2dd72f24-c335-4808-ab0a-944ef55ab21e.png)
**unicast**Then fell into an infinite loop
![Selection_009](https://user-images.githubusercontent.com/70084772/126453661-74cc6143-4381-4212-a1ff-6e2fcbf4dbef.png)
**multicast**There seems some outputs, but it is not rigid body location information
![Selection_010](https://user-images.githubusercontent.com/70084772/126453728-d0137c00-17a1-4eb0-afd1-4d1099b554c1.png)
![Selection_011](https://user-images.githubusercontent.com/70084772/126453880-b8f4da83-3e20-4783-bd5f-9a1998f778b1.png)

**This seems to be due to a problem with my motive settings，hoping get help, thanks！**
<img width="164" alt="Snipaste_2021-07-21_15-42-16" src="https://user-images.githubusercontent.com/70084772/126454149-f91553f7-a9ec-4461-be15-5613998ba448.png">
tips：It seems that my streaming settings pane is not fully display, and I can’t find where to set this.

**Comment by whoenig** [comment]
- Your NatNetSDKCrossplatform output looks actually good and you are getting data for one rigid body.
- Your settings in Motive look OK. You should be able to drag the size of the window to see the full output.
- You should check if `mocap_helper.launch` produces the desired output.
- The error you are experiencing is actually related due to communication issues with your Crazyflie, not due to issues with your mocap. Did you set the address correctly?

**Reply by Sherry-97** [reply]
Thank you for such a quick reply, I will follow your comments to check my questions

**Reply by Sherry-97** [reply]
Sorry to disturb you again. I  assign addresses to crazyflie by Crazyflie Python Client. But when I restart crazyflie after setting the address, the address displayed on the panel does not seem to change.
![Selection_002](https://user-images.githubusercontent.com/70084772/128684711-348fbb8f-ed5a-414c-b02b-95469a51fece.png)

Some related configurations
`crazyflies.yaml`
![Selection_012](https://user-images.githubusercontent.com/70084772/128685126-f3c869e1-6364-4a11-a416-bc43471101dd.png)
`mocap_helper.launch`
![Selection_013](https://user-images.githubusercontent.com/70084772/128685283-8793ecbc-c33f-41c1-9a6c-b3dae8e79ab9.png)
![Selection_014](https://user-images.githubusercontent.com/70084772/128685322-d916f09d-dd61-43fe-b2c3-b7f9aec1cc41.png)
I still encounter the previous problem, there is no image in rviz, and an error is reported
![Selection_015](https://user-images.githubusercontent.com/70084772/128685475-a52b775c-a597-4f31-8d70-5b214eee3010.png)
![Selection_016](https://user-images.githubusercontent.com/70084772/128685488-ceba0755-faac-4fd0-97f6-4cf791c547a6.png)
**And the output of running `mocap_helper.launch`**
![Selection_017](https://user-images.githubusercontent.com/70084772/128685770-83cc1adf-8619-40e0-8d4c-c0bd06d784b1.png)

points and objects are none. @whoenig

**Reply by whoenig** [reply]
Those are still two independent problems:
1.) You seem to be connected to OptiTrack, but not receiving any points/rigid bodies.
2.) There is a communication issue with your Crazyflie.

Those problems are independent and should be looked at separately. 

For 1.) What are your motive settings now? Did you define a rigid body in Motive or are you trying to use the custom tracking?
For 2.) It's best to reconfigure the Crazyflie using the cfclient while the CF is connected over USB. After reconfiguration reboot the CF and then make sure that after the reboot the data is still as expected (channel, datarate, address). Also make sure that you are on a recent firmware, e.g., the latest release.

**Reply by Sherry-97** [reply]
Thanks for your help. After I solved the two problems you mentioned, my crazyflie can take off normally!!
